### PR TITLE
5-get-articles

### PR DESCRIPTION
**Pull request by CraigAT1181**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -23,32 +42,51 @@ describe("/api/topics", () => {
...
     return request(app)
-      .get("/api/top")
-      .expect(404)
+      .get("/api/articles")
+      .expect(200)
       .then((response) => {
-        expect(response.body.msg).toBe("Path not found.");
+        const { articles } = response.body;
+        expect(articles.length).toBe(13);
+        expect(Array.isArray(articles)).toBe(true);
```
Comment (Pipe-Smoking-Rabbit) [inline __tests__/app.test.js]: If you're doing the `forEach` on the line below, then line 53 isn't necessary as it must be an array

File: __tests__/app.test.js [hunk 2]
```diff
@@ -23,32 +42,51 @@ describe("/api/topics", () => {
...
+  test("Should return sorted articles in desc order by date.", () => {
+    return request(app)
+      .get("/api/articles")
+      .expect(200)
+      .then((response) => {
+        const { articles } = response.body;
+        const latest = articles[0].created_at;
+        const earliest = articles[12].created_at;
+
+        expect(latest > earliest).toBe(true);
```
Comment (Pipe-Smoking-Rabbit) [inline __tests__/app.test.js:104]: Check out the npm package `jest sorted` as it's tailor made for asserting on the sort order of arrays and would be more effective

File: __tests__/app.test.js [hunk 3]
```diff
@@ -23,32 +42,51 @@ describe("/api/topics", () => {
...
     return request(app)
-      .get("/api")
+      .get("/api/articles")
       .expect(200)
       .then((response) => {
-        expect(typeof response.body).toBe("object");
+        const { articles } = response.body;
+        articles.forEach((article) => {
+          expect(article).not.toHaveProperty("body");
+          expect(article).toHaveProperty("comment_count");
```
Comment (Pipe-Smoking-Rabbit) [inline __tests__/app.test.js:82]: We can be much more specific than this, there are other properties we expect to also be on the article object as well as _comment_count_ , check out `toMatchObject` for a jest matcher that is useful for this type of assertion 
And `expect.any(<dataType>)` for checking the values are any _string_ or _number_ and so on

File: playground/playground.sql [hunk 4]
```diff
@@ -1,9 +1,17 @@
...
-SELECT * FROM comments;
-SELECT * FROM topics;
-SELECT * FROM users;
+SELECT
+articles.author, articles.article_id, articles.topic, articles.created_at,
+articles.votes, articles.article_img_url, COUNT(comments.comment_id)::INT AS comment_count
+FROM articles
+LEFT JOIN comments ON articles.article_id = comments.article_id
+GROUP BY articles.article_id
+ORDER BY articles.created_at DESC;
```
Comment (Pipe-Smoking-Rabbit) [inline playground/playground.sql:9]: It's great to see you've got this SQL file for playing around with query syntax to see what it does, I'd recommend adding the file names to your _.gitignore_ so they don't show up in the file changes for each PR however

File: controllers/topics.controllers.js [hunk 5]
```diff
@@ -35,3 +36,14 @@ exports.getArticleById = (req, res, next) => {
       next(err);
     });
 };
+
+exports.getArticles = (req, res, next) => {
+  fetchArticles()
+    .then((articles) => {
+      console.log({ articles });
+      res.status(200).send({ articles });
```
Comment (Pipe-Smoking-Rabbit) [inline controllers/topics.controllers.js:43]: don't forget to remove console logs once you no longer need them 🧹

File: __tests__/app.test.js [hunk 6]
```diff
@@ -23,32 +42,51 @@ describe("/api/topics", () => {
...
-      .expect(404)
+      .get("/api/articles")
+      .expect(200)
       .then((response) => {
-        expect(response.body.msg).toBe("Path not found.");
+        const { articles } = response.body;
+        expect(articles.length).toBe(13);
+        expect(Array.isArray(articles)).toBe(true);
+        articles.forEach((article) => {
+          expect(typeof article).toBe("object");
```
Comment (two-triangles) [inline __tests__/app.test.js:54]: We need to be more specific that just checking that the array contains objects. We should be writing an assertion that looks at how the object is shaped(contains the correct keys) and the correct value types for the properties of that object(check the typeof the values)

File: controllers/topics.controllers.js [hunk 7]
```diff
@@ -35,3 +36,14 @@ exports.getArticleById = (req, res, next) => {
       next(err);
     });
 };
+
+exports.getArticles = (req, res, next) => {
+  fetchArticles()
+    .then((articles) => {
+      console.log({ articles });
```
Comment (two-triangles) [inline controllers/topics.controllers.js]: remove console.logs that don't need to be there

File: models/topics.models.js [hunk 8]
```diff
@@ -32,3 +32,22 @@ exports.fetchArticleById = (article_id) => {
...
+  return db
+    .query(
+      `
+    SELECT
+articles.author, articles.article_id, articles.topic, articles.created_at,
+articles.votes, articles.article_img_url, COUNT(comments.comment_id)::INT AS comment_count
+FROM articles
+LEFT JOIN comments ON articles.article_id = comments.article_id
+GROUP BY articles.article_id
+ORDER BY articles.created_at DESC;
```
Comment (Pipe-Smoking-Rabbit) [inline models/topics.models.js:46]: This looks like a great query string!

**Review by Pipe-Smoking-Rabbit — APPROVED** [review]
Nice work on this ticket! I think the testing could be stronger especially by refactoring to use `jest sorted` so that you're checking the sort order for the whole array. I don't think you need two `forEach` loops to check the contents on the array they could be combined into one test. 
Controllers and models are looking really good, make any changes you wish to and then be free to merge 👍

**Comment by CraigAT1181** [comment]
btw - I plan on rearranging the code in the next Task so they are divided into their own controller/model files. That's why the files have been created but are currently empty.

Cheers.

**Pull request merged**